From: Don Zickus <dzickus@redhat.com>
Date: Mon, 10 May 2010 16:14:02 -0400
Subject: [virt] xen: fix pud_present compile warnings
Message-id: <1273508042-2337-1-git-send-email-dzickus@redhat.com>
Patchwork-id: 24928
O-Subject: [RHEL-6 PATCH] [misc] xen: fix pud_present compile warnings
Bugzilla: 590760
RH-Acked-by: Andrew Jones <drjones@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=590760

Working with patches that involved changing xen specific code is
never fun.  What makes it worse is when you are flooded with compiler
warnings and can't see what is going on.

One particular warning

include/asm/mach-xen/asm/pgtable.h: In function ‘pud_present’:
include/asm/mach-xen/asm/pgtable.h:102: warning: ‘pud_val’ is static but
used in inline function ‘pud_present’ which is not static

was responsible for the bulk of the problem.

A simple change from 'extern' to 'static' fixes the problem.  Not sure
why it was 'extern' to begin with.  I assume it was a typo?

Please review and ACK.

Signed-off-by: Don Zickus <dzickus@redhat.com>

diff --git a/include/asm-x86_64/mach-xen/asm/pgtable.h b/include/asm-x86_64/mach-xen/asm/pgtable.h
index ee0570d..c04eb06 100644
--- a/include/asm-x86_64/mach-xen/asm/pgtable.h
+++ b/include/asm-x86_64/mach-xen/asm/pgtable.h
@@ -99,7 +99,7 @@ extern unsigned long empty_zero_page[PAGE_SIZE/sizeof(unsigned long)];
 #define pgd_none(x)	(!pgd_val(x))
 #define pud_none(x)	(!pud_val(x))
 
-extern inline int pud_present(pud_t pud)	{ return !pud_none(pud); }
+static inline int pud_present(pud_t pud)	{ return !pud_none(pud); }
 
 static inline void set_pte(pte_t *dst, pte_t val)
 {
